src/Chapter3.hs

```diff
@@ -910,6 +981,21 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold { goldAmount :: Int } deriving (Show)
+instance Append Gold where
+  append :: Gold -> Gold -> Gold
+  append a b = Gold (goldAmount a + goldAmount b)
+
+instance Append [a] where
+  append :: [a] -> [a] -> [a]
+  append la lb = la ++ lb
+
+instance (Append a) => Append (Maybe a) where
+    append :: (Maybe a) -> (Maybe a) -> (Maybe a)
+    append (Just x) (Just y) = Just (append x y)
+    append (Just x) Nothing = Just x
+    append Nothing (Just y) = Just y
+    append Nothing Nothing = Nothing
```

Comment:
This is an absolutely correct solution!

However, we can notice that these 3 cases could be reduced to 2!
Let's see what we can unite in here.

If some of the arguments are `Nothing`, you actually always return the second one! So you need to inspect only two cases: `Nothing` in the first position, and `Nothing` on the second one 🙂